### PR TITLE
Add decryption in the module itself

### LINKED ISSUE #32: [Feature] Decryption Refactor
Decryption responsibility for LoRaRAW Unique needs to be move from Driver-LoRa to here.
Decryption needs to happen on _per device_ basis.
Requires https://github.com/NubeIO/driver-lora/issues/9 first

1. Update the device schema to accept a Hex based string as the key
    - **MUST** be 32 chars long
    - make sure it's set to `secret` so it's not displayed as plain text
2. implement decryption as in [Driver-LoRa](https://github.com/NubeIO/driver-lora/blob/master/src/loRaRaw.py#L57)
3. use the device key, revert to config `default_key` if no key exists

### PR DESCRIPTION
**Pull request by maharjanaman**
Resolves #32

### CODE REVIEW

File: schema/device.go [hunk 1]
```diff
@@ -12,6 +12,13 @@ const (
 	DeviceModelRubix        = "Rubix"
 )
 
+type DeviceKey struct {
+	Type  string `json:"type" default:"string"`
+	Title string `json:"title" default:"Device Key"`
+	Min   int    `json:"minLength" default:"32"`
```
Comment (Shiny380) [inline schema/device.go:18]: @maharjanaman I believe these have to be `maximum` and `minimum`. please confirm

**Review by Shiny380 — CHANGES_REQUESTED** [review]
@maharjanaman this is not actually applied anywhere...
you need to be applying it here: https://github.com/NubeIO/module-core-loraraw/blob/master/pkg/app.go#L108

File: schema/device.go (hunk 1 above)
Comment (maharjanaman) [reply schema/device.go:18]: For JSON, with `minLength` it's working with the UI (RCE). Are you referring to `struct` property `Min`. Have left similar comment in another comment too.
In JSON, we have `minLength` for string type and `minimum` for number type.

**Comment by maharjanaman** [comment]
> @maharjanaman this decryption is not actually applied anywhere...? you need to be applying it here: https://github.com/NubeIO/module-core-loraraw/blob/master/pkg/app.go#L108

I've added `decryptData` method and used it as shown in the attached screenshot. From the recent refactor regarding decryption in driver-lora, every type of data (unique & legacy) are being transmitted over loRa1 serial port as it is, I didn't do any checks before calling the decryption method. @Shiny380 Is this okay?

image

**Comment by Shiny380** [comment]
@maharjanaman I made some changes. I wasn't sure what the best way was to account for legacy devices so I needed to play with the code to think. I didn't test any of my code but I think this is an ok way to check for legacy devices

**Comment by maharjanaman** [comment]
@Shiny380 Here, https://github.com/NubeIO/module-core-loraraw/blob/948b4488cdd43079871914cb42f920093a198e72/pkg/app.go#L136 If the device is `nil`, do we need to proceed with decryption and decoding?

**Comment by Shiny380** [comment]
legacy devices are fully encrypted, including the address bytes at the start, so we won't find the device in the database _**without**_ first attempting to decrypt it as a legacy packet

**Comment by Shiny380** [comment]
Resolves #32

**Review by Shiny380 — APPROVED** [review]

**Pull request merged**